eng/pipelines/templates/jobs/pack.yml

```diff
@@ -55,15 +55,46 @@ jobs:
       ServiceDirectory: ${{ parameters.ServiceDirectory }}
       PackageInfoDirectory: $(Build.ArtifactStagingDirectory)/PackageInfo
 
-  - task: Powershell@2
-    displayName: "Pack Crates"
-    condition: and(succeeded(), ne(variables['NoPackagesChanged'],'true'))
-    inputs:
-      pwsh: true
-      filePath: $(Build.SourcesDirectory)/eng/scripts/Pack-Crates.ps1
-      arguments: >
-        -OutputPath '$(Build.ArtifactStagingDirectory)'
-        -PackageInfoDirectory '$(Build.ArtifactStagingDirectory)/PackageInfo'
+  - ${{ if eq('auto', parameters.ServiceDirectory) }}:
+    - task: Powershell@2
+      displayName: "Pack Crates (PR build)"
```

Comment:
Nit: quotes aren't necessary in any string unless it starts with a special YAML char or could otherwise be interpreted as number, such as versions (which will then fail with more than one `.`). Just a nit, though. I mean, you don't quote anything else here.

Reply:
And do we need `(PR build)`? It's a mutually exclusive section; but, if it helps to identify at a glance, sure.